### PR TITLE
Fix argument checks made in PR #304 and #338

### PR DESCRIPTION
**Pull request by CrazedProgrammer**
This PR fixes some faulty argument checks made in the PRs referenced above.  
These checks didn't align with the documentation/wiki and resulted in errors where they shouldn't happen.

### CODE REVIEW

File: src/main/resources/assets/computercraft/lua/rom/apis/textutils.lua [hunk 1]
```diff
@@ -329,11 +332,11 @@ function unserialize( s )
 end
 
 function serializeJSON( t, bNBTStyle )
-    if type( t ) ~= "string" then
-        error( "bad argument #1 (expected string, got " .. type( t ) .. ")", 2 )
+    if type( t ) ~= "table" or type( t ) == "string" or type( t ) == "number" or type( t ) == "boolean" then
```
Comment (dan200) [inline src/main/resources/assets/computercraft/lua/rom/apis/textutils.lua]: you clearly haven't tested this code? there's no way this line would work.

Comment (lupus590) [inline src/main/resources/assets/computercraft/lua/rom/apis/textutils.lua]: Replace all of the  `==` with `~=`

Comment (Wojbie) [inline src/main/resources/assets/computercraft/lua/rom/apis/textutils.lua]: And or with and.
`if type( t ) ~= "table" and type( t ) ~= "string" and type( t ) ~= "number" and type( t ) ~= "boolean" then`

Comment (Wojbie) [inline src/main/resources/assets/computercraft/lua/rom/apis/textutils.lua]: Sadly CrazedProgramer is on vacation right now so its not getting changed in some time.

**Comment by CrazedProgrammer** [comment]
Fixed it. I'm really sorry about this.

**Pull request merged**